### ISSUE TITLE
PNPM not supported; `Failed to resolve dependency`

### ISSUE THREAD

**Issue by nicu-chiciuc** [mui/mui-toolpad#2244]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Reproduction repo: https://github.com/nicu-chiciuc/repro_create-toolpad-app

Steps:
1. pnpm dlx create-toolpad-app repro_create-toolpad-app
2. cd ./repro_create-toolpad-app/
3. pnpm run dev

### Current behavior 😯

https://github.com/nicu-chiciuc/repro_create-toolpad-app

![Screenshot of error](https://raw.githubusercontent.com/nicu-chiciuc/repro_create-toolpad-app/main/screenshot.png)

### Expected behavior 🤔

I would like it to run.

### Context 🔦

I was trying to set it up in turborepo using pnpm, since pnpm is the recommended way in their docs.

I have a project that uses MUI and wanted to try it out.

### Your environment 🌎

```
dependencies:
  '@mui/toolpad':
    specifier: ^0.1.18
    version: 0.1.18(@babel/core@7.22.5)(@mui/base@5.0.0-beta.5)(react-dom@18.2.0)(react@18.2.0)(webpack@5.88.1)
```

OSX 12.5 (21G72)

**Comment by nicu-chiciuc** [comment]
turborepo mentioned here: https://github.com/mui/mui-toolpad/issues/1158#issuecomment-1280636184

**Comment by nicu-chiciuc** [comment]
I've also tried
`pnpm create toolpad-app my-app`

and get the same result

**Comment by Janpot** [comment]
Can confirm. It's due to our runtime trying to import from `@mui/toolpad-app` which isn't a direct dependency of the project. 

To solve this, we will combine packages `@mui/toolpad-app` and `@mui/toolpad` as `@mui/toolpad` is not much more than a wrapper for `@mui/toolpad-app` at the moment.

In the short term, if you want to use `pnpm`, you could try manually installing `@mui/toolpad-app` in your project

**Comment by nicu-chiciuc** [comment]
Adding just `@mui/toolpad-app` didn't seem to solve the issue.

To solve it, I copied all dependencies from https://github.com/mui/mui-toolpad/blob/master/packages/toolpad-app/package.json
(I assume not all are needed, but it was simpler to copy all of them).

Updated repro in the fix branch: https://github.com/nicu-chiciuc/repro_create-toolpad-app/blob/fix/package.json

**Comment by Janpot** [comment]
Yep, I've noticed that after doing https://github.com/mui/mui-toolpad/pull/2245. Looks like I can get it working by

```
pnpm add @mui/x-data-grid-pro
```

Opening https://github.com/mui/mui-toolpad/pull/2248

**Comment by nicu-chiciuc** [comment]
I've just tried your solution:
```
    "@mui/x-data-grid-pro": "^6.9.1"
```
I still get an error when running `pnpm run dev`.

branch `fix02`: https://github.com/nicu-chiciuc/repro_create-toolpad-app/blob/fix02/package.json

**Comment by Janpot** [comment]
The following works for me

```json
{
  "name": "repro_create-toolpad-app",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "toolpad dev",
    "build": "toolpad build",
    "start": "toolpad start"
  },
  "dependencies": {
    "@mui/toolpad": "https://pkg.csb.dev/mui/mui-toolpad/commit/c13d04f1/@mui/toolpad"
  }
}

```

(https://pkg.csb.dev/mui/mui-toolpad/commit/c13d04f1/@mui/toolpad is a published version of `@mui/toolpad` for pull request https://github.com/mui/mui-toolpad/pull/2248. The permanent fix will be included in `@mui/toolpad@0.1.19`)

**Comment by nicu-chiciuc** [comment]
Using 
```
  "dependencies": {
    "@mui/toolpad": "https://pkg.csb.dev/mui/mui-toolpad/commit/c13d04f1/@mui/toolpad"
  }
```
makes the project run, although I still get the warnings:
```
Failed to resolve dependency: @emotion/cache, present in 'optimizeDeps.include'
Failed to resolve dependency: @emotion/react, present in 'optimizeDeps.include'
Cannot optimize dependency: @mui/icons-material/ArrowDropDownRounded, present in 'optimizeDeps.include'
...
```

I've also tried to add this dependency to an existing turborepo with pnpm. It can also start successfully there
